### ISSUE TITLE
App-Center in Elementary/Hera,appears to have been compromised.

### ISSUE THREAD

**Issue by moxlessB3strd** [elementary/appcenter#1351]
I am running a DNS-proxy-server,in Elementary OS Hera.No problems at all;until recently.
(It may be that these started sutlely,almost as if minor changes to network-settings in this OS,were being done-I began to find small changes to the basic DNS-settings,that I'd put in place.)
Having previously used Quad 9,from IBM,as a proxy-server,I decided to try out CloudFlare's free service.
I will admit freely,that part of the problem is me-I know just enough to get into trouble.However,as you may've been hearing more and more;even the days when some of us counted on Linux being susceptible to major hacks,because it was a seemingly inconsequential OS-those days now seem to be fading fast.(It's highly possible;no one is immune,no matter how well-meaning they are..unfortunately,it is in the nature of people to form groups which will dominate other groups.)This is hardwired into human-beings,as is the inability of any given group to be "wrong".
I mention this,in my clumsy attempt to not have your group automatically take this as a criticism.
I'll now try to describe what is happening with the App-Center that is concerning me and may even be unsafe,as to the OS'es integrity.It no longer behaves as it once did: it stopped showing any and all updates,a few days ago.Then when I tinkered with my network-settings,to try to fix this,the App-Center then began to display seemingly phony updates-I call them this,because suddenly,it no longer required an administrator password at all..it has done this several times now.And the update descriptor method changed somewhat also.
But before this occurred;it was definately broken,for awhile.I notice also,that the friendly "advert" that displayed within the "Home"selection of App-Center,was entirely gone-no more offerings of the latest popular-apps,as they'd automatically scroll past,displaying each one briefly.
(That first seemed to happen,when I disabled my DNS-proxy,momentarilly;to test the system.But first,it just crashed continually and wouldn't have any updates at all.)But,Linux can be very quirky in this way;sometimes it seems as though it's ingeniously designed in an almost unconscious manner to brilliantly confuse or "Humble" the newbie.(Now,if it could fix itself in the same wonderful way-that'd be brilliant.. :)But maybe if you could laugh along with me about it;then that might more helpful than either of us could hope for,in a way.Thanks;Hoping something may come of this.

**Comment by davidmhewitt** [comment]
I'm finding this post very hard to read and understand, but, I think you're saying that AppCenter can now update packages without prompting for an admin password now, when it couldn't previously.

This is a new addition in the latest release of AppCenter that came out about a week ago. It makes the barrier to keeping your system up to date (and hence secure) that bit lower. Installing new software and removing software still requires an admin password.

The assumption as to why it's safe to allow updates without a password is that software that comes pre-installed with elementary comes from trusted sources (and is hence safe, and even recommended to update) and any additional software you have added, you had to take the explicit action to install/trust it, so updating it after that should also be fine. 

As with all computers, if you choose to install untrusted software from random sources, you run the risk of it doing bad things. For example, if you make the mistake of installing some malicious software, it's already far too late to be concerned about the fact that AppCenter is allowed to update it without a password when you click the update button.

I think you're trying to describe some other issues too, but I'm having a hard time understanding what. DNS servers and Proxy servers are very different things. If you've configured one or both, it's possible that AppCenter doesn't support the particular configuration you've applied and is hence not working as expected. If that's the case, please raise a separate issue describing clearly the issue you're experiencing and include concise instructions on how to reproduce the issue if possible.

I don't think there are any security issues here, but as a note for the future, it's generally very bad practice to report anything you suspect is a security issue on a public forum like GitHub. That applies to any software/hardware. I appreciate that isn't necessarily made clear in the issue reporting instructions, but please find a more private method of communicating potential security issues in future.

This issue report doesn't really concisely describe any particular issue in a way that we can take any action, so I'm going to close it. Please feel free to open more issues that describe individual problems you are experiencing (maybe include screenshots and instructions on how to reproduce the issue). You can also use the issue template which will help prompt you to provide more useful information.

**Comment by moxlessB3strd** [comment]
I don't know,David.But I just can't shake the feeling that Linux isn't
really "free" at all,in the sense that I must not disagree nor complain,as
a user.Why?Because it's free and we work so hard to make it so?That's what
I'm repeatedly getting back from your group..it's kind anal to expect that
the user must be perfect and must know everything-even exactly how to
complain,how to scientifically detail the cause of the problem.To actively
discourage the user from publically pointing out security-issues also
smacks of a kind of perfectionism,coupled with vagueness;because how then
am I to point these out?Your vaunted system/OS,is no longer updating on a
regular basis at all,period.If you or your group can't hear that,then
perhaps somehow you'all might need me to be completely wrong-you may wish
to take another look at this,as a group.(IE:I'm not perfect and neither
will you be.)John

**Comment by davidmhewitt** [comment]
Hi John,

I did not say you're not allowed to disagree or complain.

I explained that AppCenter is now capable of updating without an admin password as that's one of the things you reported as an issue in your post. I explained it wasn't an issue, but an intentional feature. I had a very hard time understanding what the rest of your original post was trying to explain.

Asking you to not post potential security issues is not because we want everything to appear perfect and that there are no issues. If you find a security issue and write about it in a public forum on the internet and it is an actual issue, a malicious entity can read that post and use it to exploit the vulnerability you have found (affecting many more users that just you). If instead you disclose a security issue privately, then it has chance to be fixed before it becomes public knowledge. You will find any software project will be very unhappy with you if you publicly disclose security vulnerabilities, this is not something unique to elementary.

It sounds like you still have an issue where you can't update the system at all using AppCenter, even though you've noticed that it can update itself without a password (so it's obviously working sometimes).

If you still believe this is a bug with AppCenter, please open another issue that clearly explains the problem. For example, since the issue sounds sporadic, how often does it happen? Does a reboot fix it? Do you get any error messages? If so, do they provide any details? Can you update from the terminal using `sudo apt update` when AppCenter can't? If not, what are the error messages?

Also, since you mentioned you've configured a "DNS-proxy-server", if and when you open another issue, can you please explain what that is and how you did it? Problems with updating could very well be caused by network configuration and until we know what that is, it'll be very difficult to help.

This is the kind of basic information we would like to be provided in an issue report so we have somewhere to start in investigating the issue. The issue template that you haven't used also prompts you to provide some of this information. Don't expect us to be able to decipher an "issue" about AppCenter being "compromised" where half of the post talks about "groups dominating other groups" and Linux being "ingeniously designed in an almost unconscious manner to brilliantly confuse or "Humble" the newbie".

I didn't mean to discourage you or offend you, but in re-reading my post, I don't think there's anything I wouldn't say again. If you provide little to no information, we can only provide little to no help.

**Comment by moxlessB3strd** [comment]
Okay,David-Thanks for your patience on all this.As to describing the
problem;unfortunately,I'm not very good with command-line stuff..but I
wonder if you could be open to my sending along screenshots of my
network-settings vis a' vis using CloudFlare.(1.1.1.1)
Was using Quad-9,(9.9.9.9)on my old Mac Laptop,with no problems,for quite
some time.(No more OS-updates on that rig:obsolete.)I suspect too,that the
DNS-settings in Firefox 76.0.1,may be causing a conflict also.
I'll be doing a speed test,will try testing network with Google's DNS-to
see if there's a difference.And I may have to reset my router-to see if
that changes things.
About the rambling nature of some of my previous letter-I'm thought to have
CFS/ME.This affects the brain.I apologize for this group-obsession
thing..Apparently,my own OS,doesn't get proper updates,due to bad sleep.
:>) John.

**Comment by moxlessB3strd** [comment]
Solved:using a proxy-server on Elementary. (1)Take screenshots/save
previous settings,to compare later.(2)Set proxy-settings to "Manual Proxy
Configuration",in Network..also tick the "Use This Proxy-Server for all
Protocols"box,as well.(3)In the "Advanced-Settings" panel,set MAC-Adress to
"Stable".(If you wish to change your MAC-adress.(If you want to protect
your tracks a bit more for public WiFi,you can set MAC-adress to
"Random"..this will generate a new random MAC-adress,each time.Don't forget
to reset it back to "Stable",or you may have problems,when combining this
with a DNS-proxy.)

(4)You'll have to switch-off that DNS-proxy,in your "Network-Settings",if
you need to do any updates.Use the slider-button at the top of this
panel,on the right side.(Note that any update notifications may likely not
arrive at all,while proxied.)To restore;slide the same proxy-button back to
"On".Then click "Apply",with your mouse.(5)Minor possible conflicts with
FireFox 77.01..check your Network-Settings,in its "Preferences" area.They
should be: "Autodetect-proxy setting for this network"..You should also
have:"Proxy-DNS when using Socks V5",ticked-off,in its box at bottom.That
Network-Settings,in FireFox will be at the bottom of its "Preferences"
page:click on its button,to open it.Knowing almost nothing about
network-settings,first caused that Socks V5 selection,to be missed-it
caused no end of troubles to trace this down..good luck,John (And thanks
again,David.)

**Comment by davidmhewitt** [comment]
Is the “proxy server” you’ve been configuring CloudFlare’s 1.1.1.1 service?

Like I said before, DNS servers and proxy servers are very different things.

CloudFlare 1.1.1.1 is a DNS service. If you configure it as a proxy server, it WILL break things, since it’s not a proxy server.

A DNS service is a bit like an address book and resolves domain names like elementary.io/google.com to IP addresses so that your device can contact them. By default, most home routers will assign you DNS servers that belong to your ISP meaning that your ISP can see which domain names your devices have looked up.

If you switch DNS servers, which you do in the advanced settings part of your network interface on elementary (The section doesn’t mention the word proxy at all), then your ISP no longer receives these address book lookups and CloudFlare or whichever server you choose gets them instead. This is potentially a slight improvement in privacy if you trust your new DNS servers more than your default ones.

A proxy server is a man in the middle. Instead of talking directly to elementary.io or google.com or wherever, your computer makes requests to the proxy to talk to those addresses for you on your behalf. This is sometimes useful in circumventing blocks on certain sites put in place by ISPs or governments and also somewhat hides your IP address from the remote site as its communicating with the proxy instead of your device. This doesn’t really offer any additional privacy from your ISP as any unencrypted traffic still flows through them anyway in the same way it did before.

Proxy servers are expensive to run as ALL traffic from your device has to flow through them. So if you think you’ve found a “free” proxy server, I’d be suspicious of it. Especially since ALL of your traffic will flow through it. Once again, CloudFlare is not a proxy service.

So, in summary, it sounds like you’ve mistakenly configured something in the proxy settings page of elementary that isn’t a proxy and this has broken AppCenter. In fact, I would expect it to break most things network related but perhaps Firefox is smart enough to figure out there is something wrong with the proxy configuration and ignore it. Alternatively, it may have never been using that config at all.

If what you have configured in the proxy settings page isn’t CloudFlare or similar and is actually a true proxy server, please let me know as this is something that should work with AppCenter.
